src/rgw/rgw_op.cc

```diff
@@ -4473,18 +4471,25 @@ void RGWBulkDelete::pre_exec()
 
 void RGWBulkDelete::execute()
 {
-  deleter = std::unique_ptr<Deleter>(new Deleter(store, s));
+  try {
+    deleter = std::unique_ptr<Deleter>(\
+          new RGWBulkDelete::Deleter(store, s));
+  } catch (std::bad_alloc) {
+    op_ret = -ENOMEM;
+    return;
+  }
```

Comment:
Casey's argumentation convinced me. Although there is is `strdup` where we try to recover through `-ENOMEM`, I think it's much better to exit early. I will rework the PR soon.